pkg/store/change/device/store_test.go

```diff
@@ -0,0 +1,224 @@
+	err = store2.Watch(ch)
+	assert.NoError(t, err)
+
+	change1 := &devicechange.Change{
+		NetworkChangeID: types.ID("network-change-1"),
+		DeviceID:        device1,
+		Values: []*devicechange.Value{
+			{
+				Path:  "foo",
+				Value: []byte("Hello world!"),
+				Type:  devicechange.ValueType_STRING,
+			},
+			{
+				Path:  "bar",
+				Value: []byte("Hello world again!"),
+				Type:  devicechange.ValueType_STRING,
+			},
+		},
+	}
+
+	change2 := &devicechange.Change{
+		NetworkChangeID: types.ID("network-change-2"),
+		DeviceID:        device1,
+		Values: []*devicechange.Value{
+			{
+				Path:  "baz",
+				Value: []byte("Goodbye world!"),
+				Type:  devicechange.ValueType_STRING,
+			},
+		},
+	}
+
+	// Create a new change
+	err = store1.Create(change1)
+	assert.NoError(t, err)
+	assert.Equal(t, devicechange.ID("device-1:1"), change1.ID)
+	assert.Equal(t, devicechange.Index(1), change1.Index)
+	assert.NotEqual(t, devicechange.Revision(0), change1.Revision)
+
+	// Get the change
+	change1, err = store2.Get(devicechange.ID("device-1:1"))
+	assert.NoError(t, err)
+	assert.NotNil(t, change1)
+	assert.Equal(t, devicechange.ID("device-1:1"), change1.ID)
+	assert.Equal(t, devicechange.Index(1), change1.Index)
+	assert.NotEqual(t, devicechange.Revision(0), change1.Revision)
+
+	// Append another change
+	err = store2.Create(change2)
+	assert.NoError(t, err)
+	assert.Equal(t, devicechange.ID("device-1:2"), change2.ID)
+	assert.Equal(t, devicechange.Index(2), change2.Index)
+	assert.NotEqual(t, devicechange.Revision(0), change2.Revision)
+
+	change3 := &devicechange.Change{
+		NetworkChangeID: types.ID("network-change-3"),
+		DeviceID:        device1,
+		Values: []*devicechange.Value{
+			{
+				Path:    "foo",
+				Removed: true,
+			},
+		},
+	}
+
+	// Append another change
+	err = store1.Create(change3)
+	assert.NoError(t, err)
+	assert.Equal(t, devicechange.ID("device-1:3"), change3.ID)
+	assert.Equal(t, devicechange.Index(3), change3.Index)
+	assert.NotEqual(t, devicechange.Revision(0), change3.Revision)
+
+	// For two devices
+	change4 := &devicechange.Change{
+		NetworkChangeID: types.ID("network-change-3"),
+		DeviceID:        device2,
+		Values: []*devicechange.Value{
+			{
+				Path:  "foo",
+				Value: []byte("bar"),
+				Type:  devicechange.ValueType_STRING,
+			},
+		},
+	}
+
+	// Append another change
+	err = store1.Create(change4)
+	assert.NoError(t, err)
+	assert.Equal(t, devicechange.ID("device-2:1"), change4.ID)
+	assert.Equal(t, devicechange.Index(1), change4.Index)
+	assert.NotEqual(t, devicechange.Revision(0), change4.Revision)
+
+	// Verify events were received for the changes
+	changeEvent := <-ch
+	assert.Equal(t, devicechange.ID("device-1:1"), changeEvent.ID)
+	changeEvent = <-ch
+	assert.Equal(t, devicechange.ID("device-1:2"), changeEvent.ID)
+	changeEvent = <-ch
+	assert.Equal(t, devicechange.ID("device-1:3"), changeEvent.ID)
+	changeEvent = <-ch
+	assert.Equal(t, devicechange.ID("device-2:1"), changeEvent.ID)
```

Comment:
Same comment on channel, please use switch statement to avoid blocking if something goes wrong. Same applies for all other instances of the test where you listen on channels